isis/src/hayabusa2/apps/hyb2pds4gen/main.cpp

```diff
@@ -0,0 +1,104 @@
+#include "Isis.h"
+
+#include <QDomDocument>
+#include <QDomElement>
+#include <QString>
+
+#include "Application.h"
+#include "Cube.h"
+#include "ExportDescription.h"
+#include "FileName.h"
+#include "Process.h"
+#include "ProcessExportPds.h"
+#include "ProcessExportPds4.h"
+#include "Pvl.h"
+#include "PvlKeyword.h"
+#include "PvlToXmlTranslationManager.h"
+
+using namespace std;
+using namespace Isis;
+
+void generateCSVOutput(Pvl &inputCubeLabel);
+
+void IsisMain() {
+  UserInterface &ui = Application::GetUserInterface();
+
+  // Check if input file is indeed, a cube
+  if (ui.GetFileName("FROM").right(3) != "cub") {
+    QString msg = "Input file [" + ui.GetFileName("FROM") +
+                  "] does not appear to be a cube";
+    throw  IException(IException::User, msg, _FILEINFO_);
+  }
```

Comment:
Do we need this check if we have a filter on the input? I know that IsisAml verifies the value of the input, like when there is a list of options for that parameter. Just want to make sure we don't have any unreachable code.

Reply:
Carryover from hyb1pds4gen, but yeah I think this is an unnecessary check.